demo/actions.py

```diff
@@ -631,3 +634,122 @@ def run(self, dispatcher, tracker, domain):
             dispatcher.utter_template("utter_no_more_steps", tracker)
 
         return []
+
+
+def get_last_event_for(tracker, event_type: Text, action_names_to_exclude: List[Text] = None, skip: int = 0) -> Optional[Any]:
+    """Gets the last event of a given type which was actually applied.
+
+    Args:
+        event_type: The type of event you want to find.
+        action_names_to_exclude: Events of type `ActionExecuted` which
+            should be excluded from the results. Can be used to skip
+            `action_listen` events.
+        skip: Skips n possible results before return an event.
+
+    Returns:
+        event which matched the query or `None` if no event matched.
+    """
+    #import copy
+    #to_exclude = action_names_to_exclude or []
+
+    def filter_function(e):
+        #logger.debug("e: {}".format(e))
+        #logger.debug("e.event: {}".format(e["event"]))
+        has_instance = e
+        if e["event"] == event_type:
+            has_instance = e
+            #logger.debug("  filtering event, intent name: {}".format(has_instance["parse_data"]["intent"]["name"]))
+        #excluded = (isinstance(e, ActionExecuted) and e.action_name in to_exclude)
+        excluded = (e["event"] != event_type or ((e["event"] == event_type and ((e["parse_data"]["intent"]["name"] == "domicile") or (e["parse_data"]["intent"]["name"] == "customertype")))))
+
+        return has_instance and not excluded
+
+    filtered = filter(filter_function, reversed(tracker.events))
+    for i in range(skip):
+        next(filtered, None)
+
+    logger.error("filtered returning: {}".format(filtered))
+    return next(filtered, None)
+
+
+def GetAlgoliaLink(hits, index: int):
+    doc_link = "- [" + hits[index]['hierarchy']['lvl0']
+    if (hits[index]['hierarchy']['lvl1']):
+        doc_link += "/" + hits[index]['hierarchy']['lvl1'].strip()
+        if (hits[index]['hierarchy']['lvl2']):
+            doc_link += "/" + hits[index]['hierarchy']['lvl2'].strip()
+    doc_link += "](" + hits[index]['url'] + ")"
+    return doc_link
+
+
+class ActionDocsSearch(Action):
+    def name(self):
+        return "action_docs_search"
+
+    def __init__(self) -> None:
+        self.algolia = AlgoliaAPI('BH4D9OD16A', '1f9e0efb89e98543f6613a60f847b176', 'rasa')
```

Comment:
Are these keys we should not hard-code?

Reply:
Yes, intended to do that and it is in the issue but Akela asked me to move on and stop working on this.